src-electron/validation/validation.js

```diff
@@ -291,26 +286,24 @@ async function getIntegerFromAttribute(attribute, typeSize, isSigned) {
  * Returns information about an integer type.
  * @param {*} db
  * @param {*} zapSessionId
- * @param {*} clusterRef
  * @param {*} attribType
  * @returns {*} { size: bit representation , isSigned: is signed type }
  */
-async function getIntegerAttributeSize(
-  db,
-  zapSessionId,
-  clusterRef,
-  attribType
-) {
+async function getIntegerAttributeSize(db, zapSessionId, attribType) {
   let packageIds = await queryPackage.getSessionZclPackageIds(db, zapSessionId)
-  let attribData = await queryZcl.selectNumberByNameAndClusterId(
+  const attribData = await types.getSignAndSizeOfZclType(
     db,
     attribType,
-    clusterRef,
     packageIds
```

Comment:
Hoping we don't have issues if 2 clusters define the same type name. 2 clusters can define a data type with the same name but different types and sizes.

Reply:
Wondering if keeping queryZcl.selectNumberByNameAndClusterId is safer but I will leave it to you.

Reply:
it does not work on some types , for example bitmaps